src/lib/isBIC.js

```diff
@@ -1,6 +1,7 @@
 import assertString from './util/assertString';
 
-const isBICReg = /^[A-z]{4}[A-z]{2}\w{2}(\w{3})?$/;
+// https://en.wikipedia.org/wiki/ISO_9362
+const isBICReg = /^[A-Za-z]{6}[A-Za-z0-9]{2}([A-Za-z0-9]{3})?$/;
```

Comment:
> NOTE: This is not part of the changes request. I am just offering suggestions and we can do it in a separate PR. I can also work on it after this PR get merged.

Great work here. 
I like the simplicity, but I could offer an alternative based on fact that the second part is country code addition i.e:

- Bank code (A-Z) : 4 letter code.
> Country code (A-Z) : 2 letter code.

- Location Code (0-9 or A-Z) : 2 digit code – either letters or numbers.
- Branch Code (0-9 or A-Z) : optional 3 digit code – either letters or numbers*.

->  We could leverage the array in `validISO31661Alpha2CountriesCodes` for making this a robust validator.

```javascript
// Making validISO31661Alpha2CountriesCodes exportable or just add to new util file
import { validISO31661Alpha2CountriesCodes } from './isISO31661Alpha2';
//... other code
export default function isBIC(str) {
  assertString(str);
  const countryCode = str.slice(4, 6);
  let validCountryCode = false;
  if(
    validISO31661Alpha2CountriesCodes.includes(countryCode) || 
    countryCode.toUpperCase === 'XK' // for Republic of Kosovo
   ) {
    validCountryCode = true
  }
  return isBICReg.test(str) && validCountryCode ;
}
```

CC. @tux-tn and @profnandaa

Reply:
i'm happy to make a change doing this - i think it's a good idea. i would just return false without executing the `isBICReg.test(str)` if it's not a country code - no reason to execute the test if we already know the answer.

it's not clear to me why using `includes` is useful though; is there a compatibility issue somewhere that i'm not aware of? this seems like it should be a straight, simple use of `indexOf()` - some has to call a function each time but even then why replace `some()` with `includes()` to does the exact same thing but with an extra layer of a function call.

Reply:
That is great. And I am happy that you are willing to make these changes. 

First, I would like to say that, the code above was a sample and required more modifications. 

1. If it not country code, please do that return as you have mentioned
2. Take the array of `validISO31661Alpha2CountriesCodes` from isISO31661Alpha2 and create a new file inside `lib/util`. Import for both isISO31661Alpha2 and BIC validators. 
3. You can make use of `indexOf()` instead of `includes()` and `some()`

I am happy to review that and give a go-ahead to merge this PR. 
Thanks again for your contributions @bmacnaughton.

Reply:
i think this is ready to be merged. it made sense to check that the bank ID was valid.